src/Kernel32/Kernel32+PROCESS_MEMORY_EXHAUSTION_INFO.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+
+    /// <content>
+    /// Contains the <see cref="PROCESS_MEMORY_EXHAUSTION_INFO"/> nested type.
+    /// </content>
+    public partial class Kernel32
+    {
+        /// <summary>
+        /// Allows applications to configure a process to terminate if an allocation fails to commit memory.
+        /// This structure is used by the <see cref="PROCESS_INFORMATION_CLASS"/> class.
+        /// </summary>
+        public struct PROCESS_MEMORY_EXHAUSTION_INFO
+        {
+            /// <summary>
+            /// The only valid value for <see cref="PROCESS_MEMORY_EXHAUSTION_INFO.Version"/>
+            /// </summary>
+            public const ushort PME_CURRENT_VERSION = 1;
+
+            /// <summary>
+            /// Value for <see cref="PROCESS_MEMORY_EXHAUSTION_INFO.Value"/> field to disable
+            /// the <see cref="PROCESS_MEMORY_EXHAUSTION_TYPE.PMETypeFailFastOnCommitFailure"/> feature
+            /// </summary>
+            public static UIntPtr PME_FAILFAST_ON_COMMIT_FAIL_DISABLE = UIntPtr.Zero;
+
+            /// <summary>
+            /// Value for <see cref="PROCESS_MEMORY_EXHAUSTION_INFO.Value"/> field to enable
+            /// the <see cref="PROCESS_MEMORY_EXHAUSTION_TYPE.PMETypeFailFastOnCommitFailure"/> feature
+            /// </summary>
+            public static UIntPtr PME_FAILFAST_ON_COMMIT_FAIL_ENABLE = new UIntPtr(0x1);
+
+            /// <summary>
+            /// This should be set to <see cref="PME_CURRENT_VERSION"/>
+            /// </summary>
+#pragma warning disable IDE1006 // Naming Styles
```

Comment:
These `#pragma warning`'s are really cluttering up the code. We don't have or need them anywhere else in the repo I think, as the only other place it appears is in `PROCESSOR_NUMBER` which you (@vatsan-madhavan) added recently. Can you remove them? 
Perhaps you have some IDE-level analyzers installed that are causing warnings to appear for you, but it isn't in the official build. Feel free to add an .editorconfig line somewhere to suppress the analyzer globally since we don't want it anyway.